sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessorClient.cs

```diff
@@ -0,0 +1,982 @@
+        private CancellationTokenSource RunningTaskTokenSource { get; set; }
+
+        private readonly ServiceBusReceiverClient _receiver;
+
+        /// <summary>
+        ///   The running task responsible for performing partition load balancing between multiple <see cref="ServiceBusProcessorClient" />
+        ///   instances, as well as managing partition processing tasks and ownership.
+        /// </summary>
+        ///
+        private Task ActiveReceiveTask { get; set; }
+
+        /// <summary>
+        ///   Called when a 'process message' event is triggered.
+        /// </summary>
+        ///
+        /// <param name="message">The set of arguments to identify the context of the event to be processed.</param>
+        /// <param name="session"></param>
+        ///
+        internal Task OnProcessMessageAsync(ServiceBusMessage message, ServiceBusSession session) => _processMessage(message, session);
+
+        /// <summary>
+        ///   Called when a 'process error' event is triggered.
+        /// </summary>
+        ///
+        /// <param name="eventArgs">The set of arguments to identify the context of the error to be processed.</param>
+        ///
+        private Task OnProcessErrorAsync(ExceptionReceivedEventArgs eventArgs) => _processErrorAsync(eventArgs);
+
+        /// <summary>
+        ///   The fully qualified Service Bus namespace that the consumer is associated with.  This is likely
+        ///   to be similar to <c>{yournamespace}.servicebus.windows.net</c>.
+        /// </summary>
+        ///
+        public string FullyQualifiedNamespace => _receiver.Connection.FullyQualifiedNamespace;
+
+        /// <summary>
+        ///   The name of the Service Bus entity that the consumer is connected to, specific to the
+        ///   Service Bus namespace that contains it.
+        /// </summary>
+        ///
+        internal string EntityName => _receiver.Connection.EntityName;
+
+        /// <summary>
+        ///
+        /// </summary>
+        public ReceiveMode ReceiveMode => _receiver.ReceiveMode;
+
+        /// <summary>
+        ///
+        /// </summary>
+        public bool IsSessionReceiver => _receiver.IsSessionReceiver;
+
+        /// <summary>
+        ///
+        /// </summary>
+        public uint PrefetchCount => _receiver.PrefetchCount;
+
+        /// <summary>
+        ///   Indicates whether or not this <see cref="ServiceBusProcessorClient"/> has been closed.
+        /// </summary>
+        ///
+        /// <value>
+        ///   <c>true</c> if the client is closed; otherwise, <c>false</c>.
+        /// </value>
+        ///
+        public bool IsClosed => _receiver.IsClosed;
+
+        /// <summary>
+        ///   The policy to use for determining retry behavior for when an operation fails.
+        /// </summary>
+        ///
+        internal ServiceBusRetryPolicy RetryPolicy => _receiver.RetryPolicy;
```

Comment:
We seem to leak a lot of our implementation via `internal`...  if this is just for testing, should we consider wrapping these in an `internal` method that can be called by tests and exposes all of the implementation details that tests need to avoid having the internal surface spread?

Reply:
This one can be private. The receiver properties needs to be internal until we add a new intermediate layer (which would also need to have internal properties).

Reply:
We will probably need to expose some things as internal just for tests though. Can you elaborate on what you mean by exposing in one method? Do you mean we should return a new internal type that has the properties we need? Or use out params?